app/charts/scatterplot/scatterplot.tsx

```diff
@@ -50,11 +51,33 @@ export const Scatterplot = () => {
 
   React.useEffect(() => {
     if (ref.current) {
-      select(ref.current).call(renderCircles, renderData);
+      select(ref.current)
+        .selectAll<SVGGElement, null>(".content")
+        .data([null])
+        .join(
+          (enter) =>
+            enter
+              .append("g")
+              .attr("class", "content")
+              .attr("transform", `translate(${margins.left} ${margins.top})`)
+              .call(renderCircles, renderData),
+          (update) =>
+            update
+              .call((g) =>
+                g
+                  .transition()
+                  .duration(TRANSITION_DURATION)
+                  .attr(
+                    "transform",
+                    `translate(${margins.left} ${margins.top})`
+                  )
+              )
+              .call(renderCircles, renderData),
+          (exit) => exit.remove()
+        )
+        .call(renderCircles, renderData);
```

Comment:
Nit: Seems like this logic is repeated across many files

Do you think it would be useful to have `renderAnimated(ref.current, renderCircles, renderData)` ?